[sx] Test points3d function with Windows PySide Python 2.7

Maybe i have an old version.

```
======================================================================
ERROR: test_points3d (silx.test.test_sx.SXTest)
Test points3d function
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\test\utils.py", line 196, in tearDown
    raise AssertionError("Exception occured in Qt thread:\n" + messages)
AssertionError: Exception occured in Qt thread:
Traceback (most recent call last):
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 509, in _itemAdded
    self.addRow(nodeFromItem(item), row + self._CHILDREN_ROW_OFFSET)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1369, in nodeFromItem
    node.addRow(ColormapRow(item))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 731, in __init__
    StaticRow.__init__(self, ('Colormap', None))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 225, in __init__
    super(StaticRow, self).__init__(children)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 52, in __init__
    super(BaseRow, self).__init__()
RuntimeError: You can't initialize an object twice!
Traceback (most recent call last):
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 509, in _itemAdded
    self.addRow(nodeFromItem(item), row + self._CHILDREN_ROW_OFFSET)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1369, in nodeFromItem
    node.addRow(ColormapRow(item))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 731, in __init__
    StaticRow.__init__(self, ('Colormap', None))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 225, in __init__
    super(StaticRow, self).__init__(children)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 52, in __init__
    super(BaseRow, self).__init__()
RuntimeError: You can't initialize an object twice!
Traceback (most recent call last):
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 509, in _itemAdded
    self.addRow(nodeFromItem(item), row + self._CHILDREN_ROW_OFFSET)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1363, in nodeFromItem
    specificInit(node, item)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1294, in initScatter2DNode
    node.addRow(ColormapRow(item))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 731, in __init__
    StaticRow.__init__(self, ('Colormap', None))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 225, in __init__
    super(StaticRow, self).__init__(children)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 52, in __init__
    super(BaseRow, self).__init__()
RuntimeError: You can't initialize an object twice!
Traceback (most recent call last):
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 509, in _itemAdded
    self.addRow(nodeFromItem(item), row + self._CHILDREN_ROW_OFFSET)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1363, in nodeFromItem
    specificInit(node, item)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 1294, in initScatter2DNode
    node.addRow(ColormapRow(item))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\items.py", line 731, in __init__
    StaticRow.__init__(self, ('Colormap', None))
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 225, in __init__
    super(StaticRow, self).__init__(children)
  File "C:\Users\opdau\Documents\valls\silx.git\build\lib.win-amd64-2.7\silx\gui\plot3d\_model\core.py", line 52, in __init__
    super(BaseRow, self).__init__()
RuntimeError: You can't initialize an object twice!
```

--- a/silx/gui/plot3d/_model/core.py
+++ b/silx/gui/plot3d/_model/core.py
@@ -50,9 +50,9 @@ class BaseRow(qt.QObject):
     """
 
     def __init__(self, children=()):
-        super(BaseRow, self).__init__()
         self.__modelRef = None
         self.__parentRef = None
+        super(BaseRow, self).__init__()
         self.__children = []
         for row in children:
             assert isinstance(row, BaseRow)
@@ -298,7 +298,7 @@ class ProxyRow(BaseRow):
         self._fget = WeakMethodProxy(fget)
         self._fset = WeakMethodProxy(fset) if fset is not None else None
         if fset is not None:
-            self.setFlags(self.flags(1) | qt.Qt.ItemIsEditable, 1)
+            self.setFlags(qt.Qt.ItemIsEnabled | qt.Qt.ItemIsEditable, 1)
         self._toModelData = toModelData
         self._fromModelData = fromModelData
 

--- a/silx/gui/plot3d/_model/items.py
+++ b/silx/gui/plot3d/_model/items.py
@@ -544,8 +544,8 @@ class InterpolationRow(ProxyRow):
             editorHint=modes)
 
 
-class _ColormapBaseRowMixIn(qt.QObject):
-    """Mixin class for colormap model row
+class _ColormapBaseProxyRow(ProxyRow):
+    """Base class for colormap model row
 
     This class handle synchronization and signals from the item and the colormap
     """
@@ -553,12 +553,12 @@ class _ColormapBaseRowMixIn(qt.QObject):
     _sigColormapChanged = qt.Signal()
     """Signal used internally to notify colormap (or data) update"""
 
-    def __init__(self, item):
+    def __init__(self, item, *args, **kwargs):
         self._dataRange = None
         self._item = weakref.ref(item)
         self._colormap = item.getColormap()
 
-        qt.QObject.__init__(self)
+        ProxyRow.__init__(self, *args, **kwargs)
 
         self._colormap.sigChanged.connect(self._colormapChanged)
         item.sigItemChanged.connect(self._itemChanged)
@@ -620,7 +620,7 @@ class _ColormapBaseRowMixIn(qt.QObject):
             self._sigColormapChanged.emit()
 
 
-class _ColormapBoundRow(_ColormapBaseRowMixIn, ProxyRow):
+class _ColormapBoundRow(_ColormapBaseProxyRow):
     """ProxyRow for colormap min or max
 
     :param ColormapMixIn item: The item to handle
@@ -630,9 +630,9 @@ class _ColormapBoundRow(_ColormapBaseRowMixIn, ProxyRow):
 
     def __init__(self, item, name, index):
         self._index = index
-        _ColormapBaseRowMixIn.__init__(self, item)
-        ProxyRow.__init__(
+        _ColormapBaseProxyRow.__init__(
             self,
+            item,
             name=name,
             fget=self._getBound,
             fset=self._setBound)
@@ -720,15 +720,17 @@ class _ColormapBoundRow(_ColormapBaseRowMixIn, ProxyRow):
         return super(_ColormapBoundRow, self).setData(column, value, role)
 
 
-class ColormapRow(_ColormapBaseRowMixIn, StaticRow):
+class ColormapRow(_ColormapBaseProxyRow):
     """Represents :class:`ColormapMixIn` property.
 
     :param Item3D item: Scene item with colormap property
     """
 
     def __init__(self, item):
-        _ColormapBaseRowMixIn.__init__(self, item)
-        StaticRow.__init__(self, ('Colormap', None))
+        super(ColormapRow, self).__init__(
+            item,
+            name='Colormap',
+            fget=self._get)
 
         self._colormapImage = None
 
@@ -755,6 +757,10 @@ class ColormapRow(_ColormapBaseRowMixIn, StaticRow):
         self.addRow(_ColormapBoundRow(item, name='Max.', index=1))
 
         self._sigColormapChanged.connect(self._updateColormapImage)
+
+    def _get(self):
+        """Getter for ProxyRow subclass"""
+        return None
 
     def _getName(self):
         """Proxy for :meth:`Colormap.getName`"""

--- a/silx/gui/plot3d/_model/items.py
+++ b/silx/gui/plot3d/_model/items.py
@@ -1220,8 +1220,9 @@ class Scatter2DPropertyMixInRow(object):
         """Set flags to enable/disable the row"""
         if event == items.ItemChangedType.VISUALIZATION_MODE:
             item = self.sender()
-            self.__isEnabled = item.isPropertyEnabled(self.__propertyName)
-            self.__updateFlags()
+            if item is not None:  # This occurs with PySide/python2.7
+                self.__isEnabled = item.isPropertyEnabled(self.__propertyName)
+                self.__updateFlags()
 
             # Notify model
             model = self.model()

